src/domain/valueObjects/Hash.ts

```diff
@@ -1,20 +1,19 @@
+import { isHash } from 'viem'
 import { ValueObject } from '../base/ValueObject'
 import { InvalidHashError } from '../errors/InvalidHashError'
 
 interface HashProps {
   value: string
 }
 
-const hashRegex = /^0x[a-fA-F0-9]{64}$/
-
 export class Hash extends ValueObject<HashProps> {
   private constructor(hash: string) {
     super({ value: hash })
   }
 
   static create(hash: string): Hash {
     if (typeof hash !== 'string' ||
-        !hashRegex.test(hash)) {
+        !isHash(hash)) {
```

Comment:
`isHash` [uses internally](https://github.com/wevm/viem/blob/f99fae12a056e1185180a561a9be3410e0d750a3/src/utils/hash/isHash.ts#L9) `isHex` which checks for [typeof hash === 'string'](https://github.com/wevm/viem/blob/f99fae12a056e1185180a561a9be3410e0d750a3/src/utils/data/isHex.ts#L11) so you could remove the check in line 15

Weirdly so, `isAddress` does not check it internally, so there you could keep it

Reply:
removed